File: 02. 구현/#4. 문자열 재정렬.java
//4. 문자열 재정렬 - p. 322

import java.util.*;

public class main {
    static Scanner s = new Scanner(System.in);

    public static void main(String[] args) {
        char[] data = s.nextLine().toCharArray();
        ArrayList<Character> list = new ArrayList<Character>();

        int value = 0;
        String result = "";

        for(int i=0; i<data.length; i++) {
            if(!('0'<=data[i] && data[i]<='9')) {
                list.add(data[i]);
            } else {
                value += (int)(data[i] - '0');
            }
        }
        Collections.sort(list);
        
        Iterator it = list.iterator();
        while(it.hasNext()) {
            result += String.valueOf(it.next());
        }
        
        System.out.println(result + String.valueOf(value));
    }
}

File: 02. 구현/#4. 문자열 재정렬.cpp
// 4. 문자열 재정렬 - p. 322

#include <iostream>
#include <list>
#include <iterator>
using namespace std;

bool compare(const char c1, const char c2) {
    return c1 < c2;
}

int main(void) {
    string data;
    list<char>::iterator it;
    cin >> data;

    list<char> mylist;
    int value = 0;
    for(int i=0; i<data.size(); i++) {
        if(!('0'<=data[i] && data[i]<='9')) {
            mylist.push_back(data[i]);
        } else {
            value += (data[i] - '0');
        }
    }

    mylist.sort(compare);

    string result = "";
    for(it=mylist.begin(); it!=mylist.end(); it++) {
        result += *it;
    }
    result += to_string(value);

    cout << result << endl;

    return 0;
}

